src/ordered_dict.jl

```diff
@@ -24,19 +24,34 @@ type OrderedDict{K,V} <: Associative{K,V}
     function OrderedDict()
         new(zeros(Int32,16), Array(K,0), Array(V,0), 0, false)
     end
+    function OrderedDict(n::Integer)
+        slotsz = max(16, (n*3)>>1 )
+        new(zeros(Int32,slotsz), Array(K,n), Array(V,n), 0, false)
+    end
     function OrderedDict(kv)
-        h = OrderedDict{K,V}()
+        h = OrderedDict{K,V}(length(kv))
```

Comment:
HasLength is a great idea, thanks. However, I can't see a way to use that
for 0.4.  Are infinite iterators really that common? Perhaps it is better
to fail for rare iterators than to require 0.5?

Pete

____________________
Peter M. Haverty, Ph.D.
Genentech, Inc.
phaverty@gene.com

On Wed, Jul 27, 2016 at 3:22 PM, Kristoffer Carlsson <
notifications@github.com> wrote:

> In src/ordered_dict.jl
> <https://github.com/JuliaLang/DataStructures.jl/pull/211#discussion_r72535317>
> :
>
> >      function OrderedDict(kv)
> > -        h = OrderedDict{K,V}()
> > +        h = OrderedDict{K,V}(length(kv))
>
> There already is Base.HasLength():
> https://github.com/JuliaLang/julia/blob/bcc2121dc31647fc0999d09c10df91d35f216838/base/generator.jl#L44
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaLang/DataStructures.jl/pull/211/files/6e26b3fd7faa3ad1303fec37e180b3adcfe9171c#r72535317>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AH02KxAQMCfHP-GTtvtIQxZaBHyfI-Mrks5qZ9oMgaJpZM4JVhGj>
> .
>